### PR TITLE
Fix bugzilla 24680 - [dip1000] final auto class method infers scope b…

### PR DESCRIPTION
**Pull request by dkorpel**
…ut no return

Likely for historical reasons, it's possible to enable `scope` inference but disable `return` inference, which breaks escape.d in the case of the bug report. Unify them under `scopeInprocess` to be consistent with `nogcInprocess` etc., and to disambiguate it from the function `inferScope(VarDeclaration)`.

### CODE REVIEW

File: compiler/src/dmd/func.d [hunk 1]
```diff
@@ -119,9 +119,9 @@ private struct FUNCFLAG
     bool safetyInprocess;    /// working on determining safety
     bool nothrowInprocess;   /// working on determining nothrow
     bool nogcInprocess;      /// working on determining @nogc
-    bool returnInprocess;    /// working on inferring 'return' for parameters
+    bool scopeInprocess;     /// infer `return` and `scope` for parameters
     bool inlineScanned;      /// function has been scanned for inline possibilities
-    bool inferScope;         /// infer 'scope' for parameters
+    bool dummy;              /// unused
```
Comment (dkorpel) [inline compiler/src/dmd/func.d:124]: If you wonder 'Why not just remove this?', you're in for a treat. Follow up PR coming soon.

Comment (dkorpel) [reply compiler/src/dmd/func.d:124]: Can be removed after https://github.com/dlang/dmd/pull/16752 is merged

Comment (ibuclaw) [reply compiler/src/dmd/func.d:124]: Or maybe not if older compilers are affected?

Comment (dkorpel) [reply compiler/src/dmd/func.d:124]: Between versions, there's no expectation of binary compatibility of AST nodes right?

**Comment by dkorpel** [comment]
Ping @RazvanN7

**Review by thewilsonator — CHANGES_REQUESTED** [review]

File: compiler/src/dmd/declaration.h [hunk 2]
```diff
@@ -638,12 +638,12 @@ class FuncDeclaration : public Declaration
...
-    bool returnInprocess() const;
-    bool returnInprocess(bool v);
+    bool scopeInprocess() const;
+    bool scopeInprocess(bool v);
     bool inlineScanned() const;
     bool inlineScanned(bool v);
-    bool inferScope() const;
-    bool inferScope(bool v);
+    bool dummy() const;
+    bool dummy(bool v);
```
[the remark this answers is not in the document]
Comment (dkorpel) [reply compiler/src/dmd/declaration.h]: I forgot it :)

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**